lib/ts/recipe/session/cookieAndHeaders.ts

```diff
@@ -12,106 +12,164 @@
 
 /**
  * @description clears all the auth cookies from the response
  */
-export function clearSessionFromCookie(config: TypeNormalisedInput, res: BaseResponse) {
-    setCookie(config, res, accessTokenCookieKey, "", 0, "accessTokenPath");
-    setCookie(config, res, refreshTokenCookieKey, "", 0, "refreshTokenPath");
-    setCookie(config, res, idRefreshTokenCookieKey, "", 0, "accessTokenPath");
-    res.setHeader(idRefreshTokenHeaderKey, "remove", false);
-    res.setHeader("Access-Control-Expose-Headers", idRefreshTokenHeaderKey, true);
+export function clearSession(config: TypeNormalisedInput, req: BaseRequest, res: BaseResponse, userContext: any) {
+    const transferMethod = config.getTokenTransferMethod({ req, userContext });
+
+    const tokenTypes: TokenType[] = ["access", "refresh", "idRefresh"];
+    for (const token of tokenTypes) {
+        setToken(config, req, res, token, "", 0, userContext, transferMethod);
+        if (transferMethod === "header" && hasTokenCookie(req, token)) {
+            setToken(config, req, res, token, "", 0, userContext, "cookie");
+        }
+    }
 }
 
-/**
- * @param expiry: must be time in milliseconds from epoch time.
- */
-export function attachAccessTokenToCookie(
-    config: TypeNormalisedInput,
-    res: BaseResponse,
-    token: string,
-    expiry: number
-) {
-    setCookie(config, res, accessTokenCookieKey, token, expiry, "accessTokenPath");
+export function getAntiCsrfTokenFromHeaders(req: BaseRequest): string | undefined {
+    return req.getHeaderValue(antiCsrfHeaderKey);
 }
 
-/**
- * @param expiry: must be time in milliseconds from epoch time.
- */
-export function attachRefreshTokenToCookie(
-    config: TypeNormalisedInput,
-    res: BaseResponse,
-    token: string,
-    expiry: number
-) {
-    setCookie(config, res, refreshTokenCookieKey, token, expiry, "refreshTokenPath");
+export function setAntiCsrfTokenInHeaders(res: BaseResponse, antiCsrfToken: string) {
+    res.setHeader(antiCsrfHeaderKey, antiCsrfToken, false);
+    res.setHeader("Access-Control-Expose-Headers", antiCsrfHeaderKey, true);
 }
 
-export function getAccessTokenFromCookie(req: BaseRequest): string | undefined {
-    return req.getCookieValue(accessTokenCookieKey);
+export function setFrontTokenInHeaders(res: BaseResponse, userId: string, atExpiry: number, accessTokenPayload: any) {
+    const tokenInfo = {
+        uid: userId,
+        ate: atExpiry,
+        up: accessTokenPayload,
+    };
+    res.setHeader(frontTokenHeaderKey, Buffer.from(JSON.stringify(tokenInfo)).toString("base64"), false);
+    res.setHeader("Access-Control-Expose-Headers", frontTokenHeaderKey, true);
 }
 
-export function getRefreshTokenFromCookie(req: BaseRequest): string | undefined {
-    return req.getCookieValue(refreshTokenCookieKey);
+export function getCORSAllowedHeaders(): string[] {
+    return [antiCsrfHeaderKey, HEADER_RID, authorizationHeaderKey, refreshTokenHeaderKey, idRefreshTokenHeaderKey];
```

Comment:
idRefreshTokenHeaderKey not required anymore

Reply:
Moved to a TODO in base PR